.github/workflows/check-catalyst.yaml

```diff
@@ -258,6 +279,14 @@ jobs:
         key: ${{ runner.os }}-llvm-${{ steps.llvm-hash.outputs.llvm-hash }}-default-build
         fail-on-cache-miss: True
 
+    - name: Get Cached MHLO Build
+      id: cache-mhlo-build
+      uses: actions/cache@v3
+      with:
+        path: mhlo-build
+        key: ${{ runner.os }}-mhlo-${{ steps.mhlo-hash.outputs.mhlo-hash }}-default-build
```

Comment:
Where are these build and source paths used in this action?

Reply:
The quantum dialects now depend on the MHLO libraries and headers, so they're used when compiling the dialects.

Reply:
> (rather than the compiler driver depending on both libraries say)

This is a good point and for a few reasons (including a discussion with @erick-xanadu around integrating Enzyme) I think a separate pybind-specific library that depends on the dialects and MHLO would be a better approach.

It is kind of nice to have the `quantum-lsp-server` aware of MHLO dialects, so I'll see if the CI can be set up such that `quantum-opt` does not have the MHLO dependency while `quantum-lsp-server` does. This doesn't strike me as the biggest deal though.